### DISCUSSION TITLE
Error user registration fresh installation

### DISCUSSION THREAD

**Discussion by macntech** [processone/ejabberd#3885] [Questions] [answered]
Hello community,

I setup a fresh server with ejabberd installed from repo and edited `ejabberd.yml` according to documentation. I use postgresql database. The server has been started without any issues. 

However, if I try to register a user account with `ejabberdctl register` I get a conflicting error independent the username, so the error occurs for any username choosen.

**Example**:
```bash
root@xxx:# ejabberdctl register test123 server.****.chat password

Error: conflict: User test123@server.****.chat already registered
```

I am not sure if the error code is correct and registration is somehow not working or just throwing a wrong indication and some other basic config is wrong. mod_register has been enabled.

Thanks for advise!

**Comment by licaon-kter** [comment]
OS version? ejabberd version? Put your cleaned config on https://gist.github.com and post a link.

**Reply by macntech** [reply]
OS: Ubuntu 20.04 Linux 5.4.0-122-generic
ejabberd: 22.5.0

You can find the config here: https://gist.github.com/macntech/7af6542b24a8dd968c1682c102c7c97f

**Comment by badlop** [comment]
That error message is generated here:
https://github.com/processone/ejabberd/blob/9a8a8437249792b8d230e36c4bf2855f5607cc9c/src/ejabberd_admin.erl#L606

I installed pgsql docker image, ejabberd compiled from 22.05, configured similar to yours (at least hosts, all the sql and auth and db options, acl, access_rules and api_permissions). Registration using ejabberdctl worked for me.

You can try using the erlang shell manually, with those erlang calls, maybe they show some clue about the problem:

1. When you have ejabberd running, in another shell run:

```
$ ejabberdctl debug
Erlang/OTP 25 [erts-13.0.2] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [jit:ns]

Eshell V13.0.2  (abort with ^G)
(ejabberd@localhost)1>
```

2. And now, in that erlang shell (remember the . at the end):

```
ejabberd_auth:store_type(<<"server.xxx.chat">>).
scram

ejabberd_auth:check_password_with_authmodule(<<"user1">>, <<"user1">>, <<"server.xxx.chat">>, <<"somepass">>).
false

ejabberd_auth:try_register(<<"user1">>, <<"server.xxx.chat">>, <<"somepass">>).
ok

ejabberd_auth:try_register(<<"user1">>, <<"server.xxx.chat">>, <<"somepass">>).
{error,exists}

ejabberd_auth:check_password_with_authmodule(<<"user1">>, <<"user1">>, <<"server.xxx.chat">>, <<"somepass">>).
{true,ejabberd_auth_sql}
```

**Reply by macntech** [reply]
Following results by posting the commands in the erlang shell:

```bash
(ejabberd@localhost)3> ejabberd_auth:store_type(<<"server.xxx.chat">>).
scram

ejabberd_auth:check_password_with_authmodule(<<"user">>, <<"user">>, <<"server.xxx.chat">>, <<"pw">>).
false

ejabberd_auth:try_register(<<"user">>, <<"server.xxx.chat">>, <<"pw">>).
{error,exists}

ejabberd_auth:check_password_with_authmodule(<<"user">>, <<"user">>, <<"server.xxx.chat">>, <<"pw">>).
false
```

So basically the same result, user exists somehow but no auth method and sql database is empty

**Comment by macntech** [comment, marked as answer]
I found the issue...The SQL password contained a special character "%" and has not been quoted in the yml config. After putting the password quoted it works as expected and user is in the SQL table:

```bash
User user@server.xxx.chat successfully registered
```

Therefore I guess during `ejabberdctl register` command the SQL connection could not be made as it was escaped due to the % in the password. 

The resulting error message is misleading as there is obviously no user registered. The message should contain a hint that SQL connection is broken if my assumption is correct.

Please confirm and advise.

**Comment by prefiks** [comment]
Seems to be right, i would always suggest to check ejabberd logs in cases like this, there quite often is more information that is displayed to user. Do you have just one auth mechanism enabled in your config? I think try_register should be able to show db_error when that happens, but maybe you have other auth mechanins enabled that could mask it?

**Reply by licaon-kter** [reply]
@prefiks you can see the config above
